packages/option-t/tools/package_json_rewriter/transformer/add_exports_field/ExportEntry.mjs

```diff
@@ -217,6 +217,9 @@ function constructDualPackagePathValue({ cjs, esm, dmts, dcts }) {
     //
     // see also https://nodejs.org/api/esm.html#esm_conditional_exports
     return Object.freeze({
+        // > Note that the "types" condition should always come first in "exports".
+        // https://devblogs.microsoft.com/typescript/announcing-typescript-4-7/#package-json-exports-imports-and-self-referencing
+        'types': dmts,
```

Comment:
If we introduce this change, TS4.9 fails by followings errors with these conditions

#### example code
```ts
import { createOk  } from 'option-t/PlainResult';
console.log(createOk);
```

#### Bad cases

- package.json's `type=module`
- `--moduleResolution Node16`
- The example code is contained by `*.cts`
- set `dmts` to this `types`.

##### Error Message

> error TS1479: The current file is a CommonJS module whose imports will produce 'require' calls; however, the referenced file is an ECMAScript module and cannot be imported with 'require'. Consider writing a dynamic 'import("option-t/PlainResult")' call instead.

Reply:
By this behavior, if we set `types`, tsc think that this _entrypoint_ is ESM by `d.ts` and this should be banned from cts file.

So we should not set `types` at this level.

Possibly, we can set `types` field to `default` case. But we need to investigate more.